---
output: word_document
bibliography: doc/references.bib
csl: doc/nature.csl
link-citations: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, message = FALSE, warning = FALSE)
```

```{r}
library(data.table)
library(flextable)
```

# Introduction

Genome-wide association studies (GWAS) have to date identified over 70 genetic risk loci for Alzheimer's Disease (AD).[@jansen2019; @wightman2021; @bellenguez2022] Most AD GWAS have focused on clinically diagnosed AD; however, clinically diagnosed AD is increasingly recognized as a heterogeneous disease that can reflect multiple independent or co-occurring pathological processes. Aged individuals in research autopsy cohorts often have more than one proteinopathy present in their brains at time of death.[@karanth2020]. Vascular factors, especially small-vessel disease, also contributes to cognitive decline.

does not always correspond with AD neuropathology and vice-versa.[@nelson2012]

We hypothesized that investigating the genetic architecture of multiple neuropathological endophenotypes (NPE) would:

1.  identify novel genetic risk loci for individual NPE
2.  identify common sources of genetic risk (pleiotropy) between NPE
3.  provide evidence that the effects of clinical AD genetic risk loci may be mediated by specific NPE.

# Methods

## Participants

```{r}
np <- readRDS("data/mega/mega_np.Rds")
nadrc <- np[, length(table(NACCADC))]
```

We used genotype and NPE data from four aged autopsy cohorts: National Alzheimer's Coordinating Center (NACC), the Religious Orders Study and the Memory and Aging Project (ROSMAP), Adult Changes in Thought (ACT), and the AD Neuroimaging Initiative (ADNI). All study participants were de-identified and deceased, and archival samples were exclusively used. Therefore, our study does not fall under the definition of "Human Subjects Research," according to the University of Kentucky Institutional Review Board because of NIH Exemption #4 -- involving the collection/study of data or specimens if publicly availablr, or/or recorded such that subjects cannot be identified.

This analysis used NACC data from `r nadrc` National Institute of Aging-funded Alzheimer's Disease Research Centers (ADRC). Individual ADRC use different recruitment methods and perform autopsies on-site, but participant data at each ADRC are collected using a standard form (<https://files.alz.washington.edu/documentation/np11-form.pdf>) and submitted to NACC where they are aggregated. The NACC Neuropathology (NP) data set based on the first version of this form was originally implemented in 2001[@besser2018], and this analysis uses data from then through the December 2021 freeze. Participants were excluded if they did not have autopsy data available or if they were noted in the NP data set to have at least one of 19 conditions that could potentially invalidate results. These conditions include brain tumors, severe head trauma, and fronto-temporal dementia (see **Supplementary Materials** for full list of variables used).

ROSMAP has been previously described and consists of harmonized data from two longitudinal cohorts: The Religious Orders Study (ROS) and the Memory and Aging Project (MAP).[@bennett2018] ROS began in 1994 and has recruited over 1,100 Catholic priests, nuns, and brothers across the United States. MAP started in 1997 and enrolls community members in northeastern Illinois. The ROSMAP NP data used in this study was received from Rush University researchers in January 2020.

The ACT study began in 1994 and recruited residents in the greater Seattle area aged 65 years and older without dementia at time of enrollment.21 The goal of the study has expanded to include three cohorts and continuous enrollment using the same enrollment criteria and has a current total of 4,960 participants across all three cohorts. The ACT NP data used in this study were obtained from Kaiser Permanente in May 2020.

The ADNI (<https://adni.loni.usc.edu>) was launched in 2003 as a public-private research partnership, led by Principal Investigator Michael W. Weiner, MD. The primary goal of ADNI has been to test whether serial magnetic resonance imaging (MRI), positron emission tomography (PET), other biological markers, and clinical and neuropsychological assessment can be combined to measure the progression of mild cognitive impairment (MCI) and early AD. A subset of ADNI participants undergo autopsy and receive neuropathological phenotyping. The ADNI NP data used in this study was downloaded from the ADNI website in October 2021.

## Genotype data and quality control

Quality control and inclusion/exclusion criteria closely follow that used in our previous brain arteriolosclerosis GWAS.[@shade2022] Genotype data for all cohorts have undergone imputation using the Trans-'Omics for Precision Medicine (TOPMed) Imputation Server and the TOPMed reference panel.[@taliun2021] NACC and ACT raw genotype data were obtained from the September 2020 freeze of the Alzheimer's Disease Genetics Consortium (ADGC) and subsequently imputed, while pre-imputed ROSMAP and ADNI genotype data were received from collaborators in the Hohman lab at Vanderbilt University in March 2021 and December 2021, respectively.

Quality control on genotype data was performed separately for each cohort. Firstly, only participants labeled as non-Hispanic white or similar (European, etc.) were used if available. After imputation, duplicate samples and samples without autopsy data available were removed. Genetic variants and participants were iteratively removed under no variants were missing in more than 5% of participants and no participants were missing more than 5% of variants (however, average genotype coverage was 99.7%, and no variants or participants were actually removed during this process). Then, participants with unusually high or low (+/- 3 standard deviations from mean) were removed. Finally, participants were merged with the 1000 Genomes Phase 3 cohorts[@the1000genomesprojectconsortium2012], and participants with substantial non-European ancestry, as determined by distance from EUR superpopulation centroid in principal component analysis using the first two principal components, were removed. Some participants had NP and genotype data available in more than one cohort. In these cases, records in the larger cohort were preferentially kept and discarded in other cohorts (*i.e.* NACC \> ROSMAP \> ACT \> ADNI in preference).

## Defining and harmonizing NPE for analysis

We note that there are substantial differences in the way that some NPE are collected in different cohorts, and our strategy for harmonizing was informed by practical considerations for maximizing available samples sizes given the available endophenotypes. Thus, several synthetic NPE were created by merging existing NPE within a cohort or by harmonizing categorical variables from one cohort and continuous variables from another. All code used for harmonization is available at <https://www.github.com/lincoln-shade/np_phewas>.

In total, eleven NPE were harmonized across the four cohorts used. This included eight ordinal outcomes and three binary outcomes. Ordinal outcomes harmonized were brain arteriolosclerosis, atherosclerosis in the circle of Willis, Braak Stage for neurofibrillary tangles (NFT), cerebral amyloid angiopathy, Consortium to Establish a Registry for Alzheimer's Disease (CERAD) score for neuritic plaques, diffuse amyloid-beta plaques, limbic-predominant age-related TDP-43 encephalopathy neuropathologic change (LATE-NC), and Lewy body pathology. Braak Stage ranges from zero (absent NFT) to six (diffuse NFT throughout cortex and large loss of neurons).[@braak1991] LATE-NC had four levels, with "None" indicating lack of TDP-43 proteinopathy, "Mild" indicating TDP-43 deposits in the amygdala only, "Moderate" indicating deposits in the amygdala and limbic system, and "Severe" indicating deposits in amygdala, limbic system, and neocortex. Except where noted otherwise, other ordinal NPE were coded in four levels, "None," "Mild," "Moderate," or "Severe." Diffuse amyloid plaque pathology followed the Thal staging of amyloid deposition in NACC, ACT, and ADNI. In ROSMAP, diffuse plaques are examined and quantified in five regions (midfrontal cortex, entorhinal cortex, inferior parietal cortex, and hippocampus), then scaled by each regions standard deviation and then averaged. ROSMAP participants were labeled as "None" if their averaged score was equal to zero, "Mild" if their score was higher than zero but $\le$ 0.5, "Moderate" if their score was between 0.5 and one, and "Severe" if their score was above one. These labels roughly corresponded to score quartiles.

The three binary NPE outcomes included hippocampal sclerosis (HS), gross infarctions, and microinfarcts. Participants were labeled as cases for HS if they have unilateral or bilateral HS, cases for gross infarctions in they have at least one chronic or acute gross infarction, and cases for microinfarcts if they had one or more microinfarcts in any region observed.

## Statistical Analyses

### Single-variant GWAS

Ordinal variables were analyzed using proportional odds logistic mixed-effects models using the POLMM R package,[@bi2021] while binary variables were analyzed with logistic mixed-effects models using the SAIGE R package.[@zhou2018] Covariates included age at death, sex, cohort, and the first 10 genetic principle components created using the PC-AiR method in the GENESIS R package.[@conomos2015]. Dense genetic relationship matrices (GRM) created using a pruned set of independent ($r^2 < 0.2$) were used to account for relatedness between participants. An additive mode of inheritance was assumed in all analyses. Analyses proceeded in two stages: in stage one, null models with fixed covariates and GRM were fitted using either POLMM or SAIGE. In stage 2, score tests were performed on each variant with saddle-point approximation used to calculate p-values. Variants achieving a p-value of $p < 5\times 10^{-8}$ were considered "genome-wide" significant. To identify independent risk loci, variants were clumped using PLINK 1.9[@chang] with a linkage-disequilibrium threshold of $r^2 \le 0.05$.

For NPE available in multiple cohorts, single-variant analyses were first performed in NACC participants. Lead variants with p-values exceeding a suggestive threshold of $p < 1 \times 10^{-6}$ were then tested in ROSMAP, ACT, and/or ADNI as available for attempted validation. Participants were then pooled for genome-wide mega-analysis.

To examine whether variants in the *APOE* region, defined as the region with 200 kilo-basepairs (KBP) from the start and end of the APOE coding region, independent of *APOE* $\epsilon$ haplotypes influence NPE risk, we re-analyzed Chromosome 19 using *APOE* $\epsilon$ diplotype as an additional covariate for NPE with significant association signals within the *APOE* locus in the main mega-analysis. *APOE* diplotypes were determined using the rs7412 and rs429358 variants according to the SNPedia online reference.[@apoe-s] Diplotype $\epsilon3/\epsilon3$ was used as the reference, while $\epsilon2/\epsilon2$, $\epsilon2/\epsilon3$, $\epsilon3/\epsilon4$, and $\epsilon4/\epsilon4$ were adjusted for using fixed effect indicator variables. For NPE with multiple independent ($r^2 < 0.05$) association signals in the *APOE* region in the main analysis not adjusting for *APOE* $\epsilon$ diplotype, only the most significant signal was reported in our results. Additional signals were only reported if they were significant in the *APOE*-adjusted models.

### Multivariate analyses

Because many NPE have high phenotype correlation that single-outcome association analysis cannot take into account, we sought to perform genetic association analyses that could better account for the common co-occurrence of of NPE. We first assessed the phenotype correlation of NPE used in the pooled GWAS using polychoric correlation and grouped NPE visually using a dendrogram. Principal components analysis (PCA) was then performed using the psych R package on each group of NPE. The first PC (PC1) of each group was then used as the outcome variable for GWAS using SAIGE in a two-stage analysis. In the first stage, PC1 was regression on fixed-effects covariates using the GRM to account for relatedness. Because PC1 was not normally distributed, the residuals of stage one were then rank inverse normalized and regressed on covariates and GRM again in stage two. Score tests were then performed on genetic variants and saddle-point approximation used to estimate p-values.

### Gene-based analyses

Gene start and end sites were determined using NCBI version 38 gene regions. Variants were mapped to a gene if they were located within 10 KBP of the gene's start or end sites. Using an unrelated set of the pooled participant data set, gene-based analyses were peformed using MAGMA v1.10.[@leeuw2015] The same fixed-effects covariates used in single-variant analyses were used, and the top variant principal components that accounted for 99.9% of variance in a gene's region were used to test for significance using an F-test. Genes with $p < 2.5 \times 10^{-6}$ were considered significantly associated with NPE.

### Colocalization analyses

All variants in each logistic GWAS with p-values $< 1 \times 10^{-5}$ were checked for significant quantitative trait locus (QTL) activity in the Genotype-Tissue Expression Project (GTEx) v8 European ancestry data set in 48 tissues and in ROSMAP bulk RNA-seq data in the dorsolateral prefrontal cortex. Colocalization analysis using the coloc R package was then performed on GWAS loci with significant QTLs. For ordinal variables, cut points widely used in previous studies were chosen to determine case-control proportions. A prior probability of colocalization of $1\times 10^{-5}$ was used, and a posterior probability of colocalization (PPH4) $> 80\%$ was used as a threshold for evidence of colocalization.

To investigate whether shared GWAS signals drive association between multiple NPE, colocalization analysis was also performed on loci with variants exceeding a p-value threshold of $p < 1 \times 10^{-4}$ and concordant effect direction for more than one NPE in the pooled mega-analytic GWAS.

### Conditional and mediation analyses

For genetic loci associated with multiple NPE, we considered mediation hypotheses to test whether the association of the lead variant with one NPE was mediated by another NPE. Regression analyses were run in R using the `glm` function for binary outcome traits or `polr` function in the MASS[@venables2002] package for ordinal traits. Mediation analyses were performed using the mediation[@tingley2014] R package with standard errors estimated using boot strap re-sampling with 1000 simulations. To assess the possibility of reverse causation, mediation analyses were performed with both NPE in each pair as outcomes.

# Results

```{r}
covar <- fread("data/mega/mega_np.covar")
nmega <- formatC(covar[, .N], big.mark = ",")
nnacc <- formatC(covar[FID == "0", .N], big.mark = ",")
nrosmap <- formatC(covar[FID == "1", .N], big.mark = ",")
nact <- formatC(covar[FID == "2", .N], big.mark = ",")
nadni <- formatC(covar[FID == "ADNI", .N], big.mark = ",")
```

```{r}
t1_env <- new.env()
source("code/manuscript/01_table1_participant_demographics.R",
       local = t1_env)
t1dt <- t1_env$t1dt
```

In total, `r nmega` participants across all cohorts underwent autopsy, had genotype data available, and passed quality control measures. This included `r nnacc` NACC, `r nrosmap` ROSMAP, `r nact` ACT, and `r nadni` ADNI participants. Participant demographics are shown in **Table 1**. NACC participants were younger (mean age-of-death 81 years) compared to ROSMAP (90 years) and ACT (88 years) participants, and had more balanced sex ratios, with 50% of NACC participants female vs. 68% and 58% in ROSMAP and ACT, respectively. NACC participants were also more likely to have an APOE $\epsilon 4$ allele (55%) vs. ROSMAP (25%) or ACT (28%). Across the 13 NPE harmonized across multiple cohorts, NACC and ADNI tended to show more advanced neuropathology than ROSMAP and ACT. Using CERAD neuritic plaque score as an example, 9% of NACC participants were rated as "none" for CERAD score, whereas 24% and 23% of ROSMAP and ACT participants were rated as "none," respectively. At the other extreme, 65% of NACC participants were rated as having "severe" neuritic plaque pathology, whereas only 33% and 27% of ROSMAP and ACT participants, respectively, were rated as such. This trend can be seen across **XX** of the NPE harmonized across cohorts and is shown in **Supplementary Table S2**.

## Single-variant analysis

```{r}
t2_env <- new.env()
source("code/manuscript/02_table2_mega_analysis_results.R",
       local = t2_env)
```

In the NACC analysis, no variants outside of the APOE region were significantly associated with any NPE. Variants within this region were significantly associated with Braak stage, CERAD score, CAA, and diffuse amyloid plaques. Variants reaching a suggestive significance threshold ($p < 1 \times 10^{-5}$ and replicating at a nominal significance threshold of $p < 0.05$ are shown in Table 2 below. Variants outside og the APOE region that reached a type-I error rate below $0.05$ using Benjamini-Hochberg adjustment included &&&.

In total, fourteen independent ($r^2 < 0.05$) loci had lead variants that exceeded the genome-wide significance threshold of $p < 5 \times 10^{-8}$ in the main pooled mega-analysis. Of these, four were duplicate signals in the *APOE* region, leaving ten significant signals across 13 NPE. Additionally, one locus with lead variant rs4803778 remained significantly associated with CAA in the *APOE*-adjusted analysis. Novel variant-NPE associations included a variant 12 KBP upstream of *COL4A1,* rs2000660, associated with atherosclerosis in the circle of Willis and a *PIK3R5* intronic variant, rs72807981, associated with Braak stage (**Table 3**). Loci identified that have been previously associated with NPE or dementia included *APOE* with Braak stage, CAA, CERAD score, diffuse amyloid plaques, and LATE-NC; *BIN1* with Braak stage; and *TMEM106B* with HS and LATE-NC.

## Multivariate analysis

Visual inspection of a dendrogram (**Supplementary Figure S2**) of NPE based on polychoric correlations reveals two primary clusters of NPE: an AD NPE cluster consisting of Braak stage, CERAD score, diffuse amyloid, and CAA; and a vascular NPE cluster consisting of atherosclerosis, arteriolosclerosis, gross infarcts, and microinfarcts. PC1 in the AD cluster accounted for 66% of variance in the cluster, while PC1 in the vascular cluster accounted for 50% of the variance in that cluster. Only the *APOE* region was significantly associated with the AD cluster PC1, while no loci were significantly associated with the vascular PC1.

## Gene-based analysis

## Colocalization analyses

Because CAA was the only NPE to have an associated locus in the *APOE* region with a signal independent of *APOE* $\epsilon$ diplotype, this region was exluded from NPE-NPE colocalization analysis. Pairs of NPE with the same variant reaching a p-value threshold of $p < 1 \times 10^{-5}$ with concordant effect directions included Braak stage and CERAD score in the *BIN1* region on Chromosome 2 and HS and LATE-NC in the *TMEM106B* region on Chromosome 7. Both loci showed a high probability of colocalization, with both having a PPH4 $> 99.9%$.

Multiple suggestive NPE loci showed evidence of colocalization with gene expression QTL (eQTL) in GTEx. A locus within *GRN* suggestively associated with HS ($p = 6.9 \times 10^{-8}$) colocalized with GRN expression (PPH4 $>99.9\%$), and the *TMEM106B* locus associated with HS and LATE-NC colocalized with *TMEM106B* eQTL. The CAA-associated *APOE* locus indepedent of *APOE* $\epsilon$ diplotype also colocalized with *APOE* eQTL in GTEx (PPH4 $= 95\%$) and with a methylation QTL (mQTL) in ROSMAP (PPH4 $>99.9\%$).

## Conditional and mediation analyses

We performed conditional and mediation analyses on both loci that colocalized between two NPE: *BIN1*, which colocalized with Braak stage and CERAD score; and *TMEM106B*, which colocalized with HS and LATE-NC. In unconditional analyses (the single-variant analyses discussed above), the lead *BIN1* variant, rs6733839, had similar effect sizes for both Braak stage ($p = 3.8 \times 10^{-10}$, OR $= 1.21$) and CERAD score ($p = 6.6 \times 10^{-8}$, OR $= 1.20$). However, when Braak stage was included as a covariate for CERAD, the association between rs6733839 and CERAD score was markedly reduced (OR $= 1.03$, $p = 0.47$). In contrast, the association between rs6733839 and Braak stage was less attenuated when adjusted for CERAD score (OR $= 1.14$, \$p = \$). In the mediation analysis, we found that Braak stage mediated 87% of the effect of rs633839 on CERAD score, and the post-mediation effect size was not significantly different than zero. For the *TMEM106B* analysis, we used the lead variant for HS, rs4721058, which was significantly associated with both HS (OR $= 0.67$, $p = 1.7 \times 10^{-11}$) and LATE-NC (OR $= 0.71$, $p = 1.7 \times 10^{-8}$). Mediation analysis resulted in moderate mediation (\~50%) for both HS and LATE-NC.

# Discussion

We performed GWAS and functional analyses for eleven neuropathology endophenotypes across four high-quality neuropathology cohorts with a maximum sample size of 7,463 participants. Our work builds on previous genetic association studies of NPE, Alzheimer's, and related dementias and provides another attempt to better understand the complex associations between different neuropathologies and genetic risk factors. We discovered intriguing new loci mapped to *COL4A1* and *PIK3R5* associated with atherosclerosis in the circle of Willis and Braak stage for neurofibrillary tangles, respectively. We also investigated known loci in *BIN1*, *APOE*, and *TMEM106B* to provide additional context on their associations with multiple NPE. However, our study has several important limitations, including sample size and heterogeneity in design between cohorts.

## *COL4A1* and atherosclerosis

One locus on Chromosome 12q34 with lead variant rs2000660 (MAF $= 9.9\%$ in study participants) located 12 KBP upstream of *COL4A1* was significantly associated with atherosclerosis in the circle of Willis. *COL4A1* codes for a component of collagen IV, an important component of basal lamina. The *COL4A1/COL4A2* locus has been previously found to be associated with numerous other vascular phenotypes, including peripheral artery disease[@klarin2019], coronary artery disease[@nikpay2015], stroke(cite), and arteriol stiffness(cite) through GWAS. Additionally, the locus has been associated with rare familial cerebrovascular diseases[@blevins2021] and lacunar ischemic stroke[@rannikmae2017] rs2000660 in particular has recently been found to be a risk variant for migraine.[@hautakangas2022] The relevance of the *COL4A1* locus to cerebral vascular traits is thus highly supported by previous research, and the biological role of collagen IV to cerebrovascular disease is likely related to disruption of the extracellular matrix.(cite) rs2000660 did not show evidence of colocalization in GTEx or ROSMAP, so the functional effect of this variant, if indeed it is has one, is not obvious from our results. rs2000660 was located just outside the 10 KBP window used for gene-based analysis, and the *COL4A1* and *COL4A2* genes were not nominally associated with atherosclerosis in our study ($p>0.05$). Furthermore, rs2000660 was not nominally associated with any other vascular NPE in our study. A previous GWAS in ROSMAP of cerebral atherosclerosis did not identify the *COL4A1* as a risk locus.[@vattathil2021] However, this sample size of this study was significantly smaller than the one used in the present study (1,325 vs 6,959). Indeed, rs2000660 only reached genome-wide significance in the pooled mega-analysis, though it was nominally significant in the ROSMAP-only validation analysis ($p=0.0081$).

## *PIK3R5* and Braak stage

The other identified novel genome-wide significant locus located on Chromosome 17p13 had a lead variant of rs72807981 (MAF $= 6.3\%$), an intronic variant within *PIK3R5*, which codes for a phosphatidylinositol 3-kinase involved in cell growth, motility, and survival. One study which compared differential mRNA expression in aged adults with Braak stage VI vs. non-demented controls found *PIK3R5* to be more highly expressed in cases than controls (false-discovery rate $= 0.01$).[@guennewig2021]

## *APOE* and NPE

Multiple variants in the APOE region were associated with pathognomonic AD NPE, including Braak stage, CERAD score, diffuse amyloid plaques, and CAA. Additionally, APOE was also associated with LATE-NC. This result was unsurprising and consistent with previous GWAS of NPE.[@beecham2014 @dugan2021] While multiple loci in this region were significantly associated with each of these four NPE, only one locus with lead variant rs4803778 remained significant (OR $= 1.24$, $p = 5.8 \times 10^{-12}$) when adjusting for *APOE* $\epsilon$ diplotype. rs4803778 was in high LD ($r^2 = 0.97$) with rs4803779, which was also significantly associated with CAA (OR $= 1.24$, $p = 6.7 \times 10^{-12}$) and is a lead mQTL in ROSMAP for site cg09555818 ($p = 3.3 \times 10^{-91}$), indicating that this locus may be associated with CAA through epigenetic regulation of *APOE*. However, this locus is located approximately 30 KBP downstream of the *APOE* coding end site and situated between the *APOC4*, *APOC2*, and *CLPTM1*, and is a stronger eQTL of *APOC2* than *APOE*, although it is strongly significant and colocalizes with expression of both genes in multiple tissues.

## *BIN1*, neurofibrillary tangles, and neuritic plaques

An locus approximately 30 KBP downstream of *BIN1* on Chromosome 2q14 was significantly associated with Braak stage (OR $= 1.21$, $p = 3.8 \times 10^{-10}$) and suggestively associated with CERAD score for neuritic plaques (OR $= 1.20$, $p = 6.6 \times 10^{-8}$). This locus is a known AD risk locus and is second only to *APOE* for strength of association with late-onset AD.[@bellenguez2022] However, the lead variant in this locus, rs6733839, was associated with neither diffuse amyloid plaques (OR $= 1.05$, $p = 0.15$) nor CAA (OR $= 1.02$, $p = 0.49$). Previous research has identified that *BIN1* variant rs744373 is associated with tau-PET but not amyliod-PET levels[@franzmeier2019] and that *BIN1* protein expression is correlated with neurofibrillary table pathology but not with diffuse or neuritic plaques.[@holler2014] While the *BIN1* locus was associated with both neurofibrillary tangle and neuritic plaque pathology in our sample, neuritic plaques contain dying nerve cell processes with aberrant tau fibrils identical to those seen in neurofibrillary tangles[@nelson2009], indicating that pathological process related to tau could mediate the association. We therefore performed mediation analysis to test the hypothesis that the association between the *BIN1* locus and neuritic plaques is mediated by overall tau burden, using Braak staging of neurofibrillary tangles as a proxy measurement. We found that tau burden completely mediated the association between BIN1 and neuritic plaques, as the association between *BIN1* and neuritic plaques was not longer significant after adjusting for neurofibrillary tangle pathology (OR = 1.03, p = 0.48, see **Figure X**). Our findings are therefore consistent with the hypothesis that the relationship between *BIN1* and Alzheimer's disease is mediated by tau rather than amyloid pathological processes.

## Limitations and conclusion

There are multiple limitations to the approach we took in this study. The four cohorts we used, while all high quality, used different recruitment and design considerations. NACC recruitment is handled independently at each ADRC and likely consists of a more clinically representative population, whereas MAP and ACT particpants may be more representative of the broader aged communities from which they are recruited. ROS recruits Catholic brothers and sisters and may represent a population more highly educated than the United States as a whole. These differences are reflected in the relative frequencies and distributions of NPE between cohorts, with NACC and ADNI tending to show more severe pathology than ACT and ROSMAP (**Tables 1 and S2**). Additionally, there is likely substantially heterogeneity in the way that NPE data are collected and graded both within and between studies. We review these potential limitations briefly in our previous GWAS of brain arteriolosclerosis.[@shade2022]

Additionally, while recruitment, autopsy, and genotyping of the cohorts used is ongoing, the sample sizes available for genome-wide investigation of NPE genetic risk is small relative to their clinical counterparts. A recent GWAS of AD included over 111,000 clinically diagnosed or "proxy" (based on family history) AD cases and nearly 700,000 controls. In contrast, our largest analyzed sample size for CERAD score totalled 7,447 participants, while LATE-NC totaled only 2,224. These sample reflect immense progress in autopsy cohorts from even eight years ago and will only continue to improve, but availabel sample sizes still pose limitations on power of GWAS and similar 'omics approaches to studying disease risk using NPE. A strength of our study is that data on multiple NPE is available for each participant, allowing for multivariate approaches to studying NPE. However, as our PCA-based multivariate analyses did not identify any risk loci not identified in single-outcome analyses, this approach cannot overcome sample size limitations.

In conclusion, we have identified several intriguing novel loci associated with NPE and validated multiple known risk loci for AD and related dementias in NPE. We also provide additional context and consideration for relationships between specific risk loci and different NPE. Overall, our study provides additional potential avenues of investigation into the relationship between genomics, neuropathology, and dementia.

\newpage

# References

::: {#refs}
:::

\newpage

# Tables

```{r}
t1_env$t1
```

 

```{r}
t2_env$t2
```

 
